Compile dist folder

nameofname opened the pull request:
Builds on top of work to create a better dist folder

nameofname commented:
Replaces https://github.com/1stdibs/dibs-vg/pull/16

nameofname commented:
@twhid this replaces old PR to create template string'd files in dist folder. Also adds the functionality to compile SVGs into react components so theoretically we would no longer need to use the react-svg-loader to load files in webpack.

nameofname commented on htmlToReact.js:
```diff
@@ -0,0 +1,48 @@
+    return new Promise((fulfill, reject) => {
+        parser.parseString(html, (err, obj) => {
+
+            if (err) { return reject(err); }
```
TODO : stop the build process if error caught.

hellatan commented:
looks more or less fine to me. i don't see anything glaring about it.

twhid commented on generateDist.js:
```diff
@@ -0,0 +1,28 @@
+    const stringContents = ["module.exports = `", fileContents, "`;", "\n"].join('');
+
+    // write output to file system
+    fs.writeFileSync((stringDest + '/' + name + '.es.js'), stringContents, { encoding : 'UTF-8' });
```
This would be cleaner with template strings IMHO, eg:
```js
fs.writeFileSync(`${stringDest}/${name}.es.js`, contents, { encoding : 'UTF-8' });
```

nameofname replied:
👍 much cleaner.

twhid commented:
@nameofname if you want this merged today please remove the "don't merge" label (I have bookmarks that exclude PRs with that label so I'm less likely to see it if it has that label).

twhid commented on generateDist.js:
```diff
@@ -0,0 +1,28 @@
+const fs = require('fs');
+const htmlToReact = require('./htmlToReact');
+const svgs = fs.readdirSync(process.cwd() + '/src');
+const stringDest = process.cwd() + '/dist/templateString';
```
moar template literals..!
```js
const stringDest = `${process.cwd()}/dist/templateString`;
```

nameofname commented:
@twhid I updated template strings as requested and added error handling to stop prepublish process if errors are caught following the advice of NPM documentation : 
https://docs.npmjs.com/misc/scripts#exiting

twhid commented on generateDist.js:
```diff
@@ -0,0 +1,45 @@
+    const name = file.split('.')[0];
+
+    fs.readFile(leFile, (err, data) => {
+        if (err) die(err);
```
Using our linter? Should be:
```js
if (err) {
    die(err);
}
```

In the `writeFile` callback too.

hellatan replied:
@twhid i added linting - https://github.com/1stdibs/dibs-vg/pull/18

twhid commented on generateDist.js:
```diff
@@ -0,0 +1,45 @@
+};
+
+svgs.forEach(file => {
+    const leFile = process.cwd() + '/src/' + file;
```
Missed this one if you're settling on string template literals.

twhid commented on generateDist.js:
```diff
@@ -0,0 +1,45 @@
+            });
+
+        }).catch(err => {
+            if (err) die(err);
```
Need to check for the truthiness of `err` here? Don't you want to exit abnormally if you're in the catch?

nameofname replied:
It will only catch if an error is returned, which would happen on Promis->reject in htmlToReact.js

twhid commented on generateDist.js:
```diff
@@ -0,0 +1,45 @@
+const reactDest = `${process.cwd()}/dist/react`;
+// a function to end process to prevent prepublish from completing if an error is encountered :
+const die = err => {
+    console.error(err);
```
As of node 4.4.4, if this is indeed an instance of Error, it will only log:
```
[Error: message]
```

You probably want to `throw` so you get the entire stack;

twhid commented on generateDist.js:
```diff
@@ -0,0 +1,47 @@
+                checkError(err);
+            });
+
+        }).catch(err => {
```
my pet peeve... `catch(checkError)` 😵

nameofname commented:
@twhid rebased and linted, ready to go.

twhid commented on generateList.js:
```diff
@@ -1,14 +1,23 @@
+    fs.readdir(`${__dirname}/src`, (err, ls) => {
+        checkError(err);
+        fs.writeFile(`${__dirname}/list.json`, JSON.stringify(ls), { encoding : "UTF-8" }, err => {
+            checkError(err);
```
`checkError` seems redundant now
```js
fs.writeFile(filename, stuff, { encoding : "UTF-8" }, err => {
    if (err) {
        throw err;
    }
});
// or, if you must
fs.writeFile(filename, stuff, { encoding : "UTF-8" }, checkErr);
// I'd also save the options..
const opts = { encoding : "UTF-8" };
// then
fs.writeFile(filename, stuff, opts, checkErr);
```

nameofname replied:
If i remove it I have to do a similar of statement twice, would rather leave as is.

twhid replied:
Pet peeve still applies. Plus I really hate long lines. 🐛 🐱
